spec/unit/sensu_event_spec.rb

```diff
@@ -0,0 +1,216 @@
+require 'spec_helper'
+require 'puppet/type/sensu_event'
+
+describe Puppet::Type.type(:sensu_event) do
+  let(:default_config) do
+    {
+      name: 'checkalive for test',
+    }
+  end
+  let(:config) do
+    default_config
+  end
+  let(:event) do
+    described_class.new(config)
+  end
+
+  it 'should add to catalog with raising an error' do
+    catalog = Puppet::Resource::Catalog.new
+    expect {
+      catalog.add_resource event
+    }.to_not raise_error
+  end
+
+  it 'should require a name' do
+    expect {
+      described_class.new({})
+    }.to raise_error(Puppet::Error, 'Title or name must be provided')
+  end
+
+  it 'should handle composite title check' do
+    expect(event[:check]).to eq('checkalive')
+  end
+
+  it 'should handle composite title entity' do
+    expect(event[:entity]).to eq('test')
+  end
+
+  it 'should allow ensure present' do
+    event[:ensure] = 'present'
+    expect(event[:ensure]).to eq(:present)
+  end
+
+  it 'should allow ensure resolve' do
+    event[:ensure] = 'resolve'
+    expect(event[:ensure]).to eq(:resolve)
+  end
+
+  it 'should allow ensure delete aliased to absent' do
+    event[:ensure] = 'delete'
+    expect(event[:ensure]).to eq(:absent)
+  end
+
+  it 'should allow ensure absent' do
+    event[:ensure] = 'absent'
+    expect(event[:ensure]).to eq(:absent)
+  end
+
+  it 'should not allow unsupported ensure' do
+    expect {
+      event[:ensure] = 'foo'
+    }.to raise_error(Puppet::Error, /Invalid value "foo"/)
+  end
+
+  defaults = {
+    'organization': 'default',
+    'environment': 'default',
+  }
+
+  # String properties
+  [
+    :organization,
+    :environment,
+  ].each do |property|
+    it "should accept valid #{property}" do
+      config[property] = 'foo'
+      expect(event[property]).to eq('foo')
+    end
+    if default = defaults[property]
+      it "should have default for #{property}" do
+        expect(event[property]).to eq(default)
+      end
+    end
+  end
+
+  # String regex validated properties
+  [
+  ].each do |property|
+    it "should not accept invalid #{property}" do
+      config[property] = 'foo bar'
+      expect { event }.to raise_error(Puppet::Error)
```

Comment:
This should probably have an error message that is expected

Reply:
Will update.

Reply:
Actually this block is empty, so the test never actually gets executed.  I left these empty test blocks around in case properties got added that fit the block's test criteria.  There are several places in code these blocks are actually tested, I'll update those in separate pull request.